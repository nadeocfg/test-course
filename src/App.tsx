import React, { useEffect, useState } from "react";

const App = () => {
  const [isVisible, setIsVisible] = useState(false);
  const [asyncData, setAsyncdata] = useState(null);
  const [header, setHeader] = useState("");

  useEffect(() => {
    setTimeout(() => {
      setAsyncdata({});
    }, 300);
  }, []);

  const onClick = () => {
    setIsVisible(!isVisible);
  };

  return (
    <main>
      <h1>Hello World!!!</h1>

      {asyncData && <div>Async data here</div>}

      <hr />

      <button onClick={onClick} data-testid="visible-btn">
        Click me
      </button>
      {isVisible && <div data-testid="visible-el">Element is visible</div>}

      <hr />

      <input
        onChange={(event) => setHeader(event.target.value)}
        value={header}
        type="text"
        placeholder="Input something here..."
      />

      <h2 data-testid="header">{header}</h2>
    </main>
  );
};

export default App;
