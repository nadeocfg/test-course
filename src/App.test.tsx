import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

describe("Tests for <App /> component", () => {
  test("Render elements", () => {
    render(<App />);
    const helloWorld = screen.getByText(/hello world/i);
    const button = screen.getByRole("button");
    const input = screen.getByRole("textbox");

    expect(helloWorld).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test("test Async data", async () => {
    render(<App />);

    screen.debug();

    const asyncEl = await screen.findByText(/async data/i);

    screen.debug();

    expect(asyncEl).toBeInTheDocument();
  });

  test("test onClick event", () => {
    render(<App />);

    const btn = screen.getByTestId("visible-btn");

    expect(screen.queryByTestId("visible-el")).toBeNull();

    fireEvent.click(btn);

    expect(screen.queryByTestId("visible-el")).toBeInTheDocument();

    fireEvent.click(btn);

    expect(screen.queryByTestId("visible-el")).toBeNull();

    fireEvent.click(btn);

    expect(screen.queryByTestId("visible-el")).toBeInTheDocument();
  });

  test("test Input event", () => {
    render(<App />);

    const input = screen.getByRole("textbox");

    expect(screen.getByTestId("header")).toContainHTML("");

    fireEvent.input(input, {
      target: {
        value: "asd bkb",
      },
    });

    expect(screen.getByTestId("header")).toContainHTML("asd bkb");
  });
});
