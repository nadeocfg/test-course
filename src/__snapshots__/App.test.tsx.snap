// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <App /> component Render elements 1`] = `
<input
  placeholder="Input something here..."
  type="text"
  value=""
/>
`;
